qa/rpc-tests/test_framework/mininode.py

```diff
@@ -994,14 +995,16 @@ def deserialize(self, f):
         self.message = deser_string(f)
         self.code = struct.unpack("<B", f.read(1))[0]
         self.reason = deser_string(f)
-        if (self.message == "block" or self.message == "tx"):
+        if (self.code != self.REJECT_MALFORMED and
+                (self.message == "block" or self.message == "tx")):
             self.data = deser_uint256(f)
 
     def serialize(self):
         r = ser_string(self.message)
         r += struct.pack("<B", self.code)
         r += ser_string(self.reason)
-        if (self.message == "block" or self.message == "tx"):
+        if (self.code != self.REJECT_MALFORMED and
+                (self.message == "block" or self.message == "tx")):
```

Comment:
Well it is important to avoid unexpected results. b'123'=='123' evaluates to False in Python3.